server/mongodb/actions/User.ts

```diff
@@ -10,3 +10,26 @@ export const createUserEmail = async (email: string): Promise<IUser> => {
   const user: IUser = await User.create({ email });
   return user;
 };
+
+export const patientSignUp = async (
+  data: SignupData,
+): Promise<IUser | null> => {
+  const result = await User.findOneAndUpdate<IUser>(
+    { email: data.email },
+    {
+      $set: {
+        name: data.name,
+        phoneNumber: data.phoneNumber,
+        patientDetails: {
+          birthDate: data.birthDate,
+          secondaryContactName: data.secondaryContactName,
+          secondaryContactPhone: data.secondaryContactPhone,
+        },
+        "patientDetails.signedUp": true,
```

Comment:
This causes an error because `patientDetails.signedUp: true` creates a new Object and tries to assign this to `patientDetails`. However, we are making another assignment to `patientDetails` in the same query, so this causes a conflict. Instead, put `signedUp: true,` within the `patientDetails` object.